public/app/plugins/datasource/prometheus/querybuilder/components/metrics-modal/state/helpers.ts

```diff
@@ -0,0 +1,213 @@
+import { AnyAction } from '@reduxjs/toolkit';
+
+import { PrometheusDatasource } from 'app/plugins/datasource/prometheus/datasource';
+import { getMetadataHelp, getMetadataType } from 'app/plugins/datasource/prometheus/language_provider';
+
+import { regexifyLabelValuesQueryString } from '../../../shared/parsingUtils';
+import { QueryBuilderLabelFilter } from '../../../shared/types';
+import { PromVisualQuery } from '../../../types';
+import { HaystackDictionary, MetricData, MetricsData, PromFilterOption } from '../types';
+
+import { MetricsModalMetadata, MetricsModalState, stateSlice } from './state';
+
+const { setFilteredMetricCount } = stateSlice.actions;
+
+export async function setMetrics(
+  datasource: PrometheusDatasource,
+  query: PromVisualQuery,
+  initialMetrics?: string[]
+): Promise<MetricsModalMetadata> {
+  // metadata is set in the metric select now
+  // use this to disable metadata search and display
+  let hasMetadata = true;
+  const metadata = datasource.languageProvider.metricsMetadata;
+  if (metadata && Object.keys(metadata).length === 0) {
+    hasMetadata = false;
+  }
+
+  let nameHaystackDictionaryData: HaystackDictionary = {};
+  let metaHaystackDictionaryData: HaystackDictionary = {};
+
+  // pass in metrics from getMetrics in the query builder, reduced in the metric select
+  let metricsData: MetricsData | undefined;
+
+  metricsData = initialMetrics?.map((m: string) => {
+    const type = getMetadataType(m, datasource.languageProvider.metricsMetadata!);
+    const description = getMetadataHelp(m, datasource.languageProvider.metricsMetadata!);
+
+    // string[] = name + type + description
+    const metaDataString = `${m} ${type} ${description}`;
```

Comment:
also, do we need `type` here? since we already have a dedicated filter?

https://user-images.githubusercontent.com/109082771/235507473-b2942eed-ccc5-4cc0-b178-c848ecf8c6f4.png

Reply:
This makes sense, having type in s a select and in the fuzzy search search is redundant. I think removing the type from the metadata search will require some more UI discussion. I think I could see the column order changing, removing highlighting from the type and so I will add this to my list of things to discuss with UI for the next round of updates. Thank you Leon!

Reply:
@leeoniya what is the benefit of `¦` over `|`?

Reply:
> what is the benefit of `¦` over `|`?

much less likely to exist in either `m` or `description` in case we want to ever assert which segment the matched terms occured.

Reply:
Received an answer on slack, ¦ is less likely to occur in metadata and will help with part matching.